juju/juju
# DB accessor worker reads charm-set configuration when change detected

This adds a new type for reading charm-set configuration from disk. It is supplied to the `dbaccessor` worker via its config.

When the worker detects via the config socket that configuration has changed, it uses the reader to acquire cluster configuration.

At this stage, the configuration is only read, and the clustering behaviour that relies on the `peergrouper` messages is unchanged. The patch to follow will finally remove the dependency on such messages.

Included is a drive-by to fix some missing commas in schema DDL.

## QA steps

- Bootstrap with an object store configuration that allows entry into HA.
- `juju enable-ha`.
- One settled, you should see controller log entries indicating acquired configuration such as:
```
machine-2: 14:04:32 INFO juju.worker.dbaccessor new cluster config: map[controller/0:10.246.27.35 controller/1:10.246.27.110 controller/2:10.246.27.20]
```

## Documentation changes

None.

## Links

**Jira card:** [JUJU-5308](https://warthogs.atlassian.net/browse/JUJU-5308)



[JUJU-5308]: https://warthogs.atlassian.net/browse/JUJU-5308?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
